packages/zquery/src/types.ts

```diff
@@ -12,6 +15,58 @@ export interface ZQueryState<DataType, FetchArgs extends unknown[] = []> {
   };
 }
 
+export type UseHookOptions<
+  ResolvedDataType,
+  SelectorType extends
+    | ((zQueryState: AbridgedZQueryState<ResolvedDataType>) => unknown)
+    | undefined = undefined,
+> = SelectorType extends undefined
+  ?
```

Comment:
Unfortunately, TypeScript (or at least, VSCode) requires that the doc blocks above these properties be repeated for each branch of this ternary operator — that is, once for when they're undefined, and once for when they're defined. 🤷🏻